templates/tapestry.js

```diff
@@ -1374,329 +1384,28 @@ function tapestryTool(config){
-                                    'muted': h5pVideo.isMuted(),
-                                    'caption': h5pVideo.getCaptionsTrack(),
-                                    'quality': h5pVideo.getQuality(),
-                                    'playbackRate': h5pVideo.getPlaybackRate(),
-                                    'time': h5pVideo.getCurrentTime()
-                                };
-    
-                                // Update the mediaButton icon to play icon
-                                updateMediaIcon(id, mediaType, 'play');
-                                seeked = true;
-    
-                                recordAnalyticsEvent('user', 'pause', 'h5p-video', id, {'time': h5pVideo.getCurrentTime()});
-                                break;
-    
-                            case h5pObj.Video.BUFFERING:
-    
-                                // Update the mediaButton icon to loading icon
-                                updateMediaIcon(id, mediaType, 'loading');
-                                break;
-                        }
-                    });
-    
-                    // Auto-play
-                    setTimeout(function(){
-                        h5pVideo.play();
-                        recordAnalyticsEvent('app', 'auto-play', 'h5p-video', id);
-                    }, 1000);
-                }
 
-            }, false);
-        } else if (mediaFormat === "embed") {
-            mediaEl = $('<iframe id="embed" src="' + mediaUrl + '" width="' + width + '" height="' + height + '" frameborder="0" allowfullscreen="allowfullscreen"><\/iframe>');
-        }
-    
-        return mediaEl;
-    }
-    
-    function createTextNodeElement(title, str) {
-        var lightboxContent = document.createElement("div");
-        var titleSection = document.createElement("div");
-        titleSection.setAttribute("id", "text-light-box-title");
-    
-        var titleText = document.createElement("h3");
-        titleText.appendChild(document.createTextNode(title));
-        titleSection.append(titleText);
-        lightboxContent.append(titleSection);
-    
-        if (str) {
-            var paragraphSection = document.createElement("div");
-            paragraphSection.setAttribute("id", "text-light-box-paragraph");
-            paragraphArray = str.split("\n\n");
-            for (var i = 0; i < paragraphArray.length; i++) {
-                var paraDiv = document.createElement("div");
-                var para = document.createElement("p");
-                para.setAttribute("id", "text-light-box-paragraph-text");
-                para.innerHTML = paragraphArray[i].replace('\n','<br>');
-                paraDiv.appendChild(para);
-                paragraphSection.appendChild(paraDiv);
-            }
-            lightboxContent.appendChild(paragraphSection);
-        }
-        return $(lightboxContent);
+    this.saveVideoProgress = function(id, currentTime, duration) {
```

Comment:
Same as my previous comment, please see if you could move this out of `tapestry.js` since it's not used anywhere else in this file.

Reply:
Unfortunately this works with d3 so it has to be kept in tapestry.js until we eventually move tapestry into a vue component.

Reply:
Only `updateViewedProgress()` works with d3, so I think it's better if we move the `updateViewedValue` function out of tapestry.js and into TapestryAPI and then use it in Vue directly. We can make the `updateViewedProgress()` accessible from outside so that we can just call that function from Vue. 

Again, I'm trying to make sure we move as much stuff out of this file as possible so that it becomes more realistic to refactor this into a component, otherwise it will remain too big for any one person to attempt to turn it into a component.